### ISSUE TITLE
Crashing on World Load (Seems related to SodiumWorldRenderer)

### ISSUE THREAD

**Issue by iancbogue** [Someone-Else-Was-Taken/Periodic-Table-Reforged#117]
I started experiencing this error last night when I joined one of my worlds and some blocks weren't rendering. Here are [two images](https://ibb.co/album/k4ZhQP) I got from the Seamless Loading Screen Mod that show what I see in-game before the crash. I'm including my crash report and debug log below. I did try to figure out what is causing it but couldn't come to a conclusion.

[crash-2021-11-18_20.06.28-client.txt](https://github.com/Someone-Else-Was-Taken/Periodic-Table-Reforged/files/7567718/crash-2021-11-18_20.06.28-client.txt)
[debug.log](https://github.com/Someone-Else-Was-Taken/Periodic-Table-Reforged/files/7567719/debug.log)

**Comment by NighthowlerStudios** [comment]
This looks like the mod Seamless Loading Screen loads chunks itself using its own chunkloading pipeline to take a screenshot on its own, which of course is dangerous as magnesium might be trying to use some optimization methods which SLS is not expecting.  I'm not sure that these two mods could ever be compatible.

**Comment by SpinHarmony** [comment]
Seamless Loading Screen doesn’t load chunks at all, it just shows a screenshot taken from the last time the world was saved as the image for the loading screen.

**Comment by NighthowlerStudios** [comment]
The path trace is showing otherwise?  are you the dev of the mod?

**Comment by iancbogue** [comment]
I tried disabling Seamless Loading Screen. It still happened.

[crash-2021-11-20_21.13.01-client.txt](https://github.com/Someone-Else-Was-Taken/Periodic-Table-Reforged/files/7575778/crash-2021-11-20_21.13.01-client.txt)
[debug.log](https://github.com/Someone-Else-Was-Taken/Periodic-Table-Reforged/files/7575779/debug.log)

[Minecraft Magnesium Crash After Removing SLS](https://youtu.be/j8L5I2XBfYk)

**Comment by NighthowlerStudios** [comment]
the rendering seems entirely broken here, which might be causing the eventual crash.  can you try to sort through your mod list to see if removing any reverse the issue (try to find which specific mod is colliding with and causing the issue)

**Comment by SpinHarmony** [comment]
I don’t use the mod. I just read the Curseforge page about what it’s supposed to do. Maybe it’s outdated.

**Comment by NighthowlerStudios** [comment]
> I don’t use the mod. I just read the Curseforge page about what it’s supposed to do. Maybe it’s outdated.

thankfully that's irrelevant now anyway as it seems something is causing magnesium to be unable to read the chunks from the internal server and thus can't render them and eventually crashes.

**Comment by NighthowlerStudios** [comment]
hmm I see, were you using Biomes o Plenty world type by any chance when recording that footage?  that mod makes an entirely new world format.  it's possible magnesium is having trouble reading chunks from non-vanilla formats.

**Comment by iancbogue** [comment]
Yes, I am using Biomes O Plenty. I believe I tried it with a normal world type as well, but I will try it again just in case I didn’t. If that doesn’t work, I’ll see about sorting through my mods again, though I won’t be able to do it today.

**Comment by orblazer** [comment]
Hello,
I have the same issue but i doesn't have Biome'O Plenty but another mod for custom generation.

[crash-2021-11-26_17.54.11-client.txt](https://github.com/Someone-Else-Was-Taken/Periodic-Table-Reforged/files/7610028/crash-2021-11-26_17.54.11-client.txt)

**Comment by NighthowlerStudios** [comment]
So yeah from what I'm gathering here, Sodium isn't able to capture chunks properly from modded generation rules because they save chunk data in a format that isn't vanilla compatible.  This results in unpredictable behaviour when trying to build the mesh as it might not be able to see the blocks correctly, and in most cases this can result in a crash down the line.

**Comment by NighthowlerStudios** [comment]
I suggest the author of this issue should rename it to "Custom World Generator mods not having chunks rendering properly, sometimes crashes"
